tokio/src/runtime/scheduler/current_thread.rs

```diff
@@ -290,6 +290,10 @@ impl Context {
     fn park(&self, mut core: Box<Core>, handle: &Handle) -> Box<Core> {
         let mut driver = core.driver.take().expect("driver missing");
 
+        let before = handle
+            .shared
+            .woken
+            .load(std::sync::atomic::Ordering::Relaxed);
```

Comment:
Please add an import.
```suggestion
        let before = handle
            .shared
            .woken
            .load(Relaxed);
```

Reply:
Will do, will add a `use`. I don't know how to quantifiably test that the park is being skipped. I can add a test where a future is blocked where the on_park callback makes it ready but I don't know how to show that the thread was not parked. I can think about it more and attempt to find something soon.

My bigger concern is I don't know how to test for potential other negative consequences to current_thread scheduling and I haven't had the time to fully understand how the scheduler works to be able to think things through. For example, the fact that some awakes happen normally during scheduling and they should *not* be handled until after a park really surprised me, at least that's what the failing unit tested indicated, so what else would be a surprise?

Once I was in a position to test performance again, I was going to play with the idea of just spawning a no-op local task in the on_park callback to see if that extra work would just be lost in the noise anyway. Then no change Tokio would be necessary.

Reply:
For tests, the best way to check if park was called is using runtime metrics (which has a park counter).